### PR TITLE
[MDS-361] Modal v2

### PR DESCRIPTION
**Pull request by tkullisaar**

### CODE REVIEW

File: lib/moon/design/button.ex [hunk 1]
```diff
@@ -26,6 +26,7 @@ defmodule Moon.Design.Button do
   prop(form, :string)
   prop(testid, :string)
   prop(on_click, :event)
+  prop(close, :event)
```
Comment (phgrey) [inline lib/moon/design/button.ex:29]: why is it here?

**Review by phgrey — APPROVED** [review]
good job. see minor comments

File: lib/moon/design/button.ex (hunk 1 above)
Comment (tkullisaar) [reply lib/moon/design/button.ex]: there by accident, removed

File: lib/moon/design/modal/backdrop.ex [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  use Surface.Component, slot: "backdrop"
+
+  import Moon.Helpers.MergeClass
+
+  prop(class, :css_class)
+
+  def render(assigns) do
+    ~F"""
+    <div class={merge([
```
[the remark this answers is not in the document]
Comment (tkullisaar) [reply lib/moon/design/modal/backdrop.ex]: Please see changes, is this the proper way?

File: lib/moon/design/modal/panel.ex [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  import Moon.Helpers.MergeClass
+
+  prop(class, :css_class)
+
+  slot(default)
+
+  def render(assigns) do
+    ~F"""
+    <div class={merge([
```
[the remark this answers is not in the document]
Comment (tkullisaar) [reply lib/moon/design/modal/panel.ex]: Please see changes, is this the proper way?

File: lib/moon/design/modal/backdrop.ex [hunk 4]
```diff
@@ -0,0 +1,13 @@
+defmodule Moon.Design.Modal.Backdrop do
+  @moduledoc false
+
+  use Surface.Component, slot: "backdrop"
+
+  prop(class, :css_class)
+
+  def render(assigns) do
+    ~F"""
+    <div class={"fixed inset-0 bg-popo opacity-[.6]", @class} />
```
Comment (phgrey) [inline lib/moon/design/modal/backdrop.ex:10]:
```
class={merge(["fixed inset-0 bg-popo opacity-[.6]", @class])}
```

File: lib/moon/design/modal/panel.ex [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  slot(default)
+  slot(header)
+
+  def render(assigns) do
+    ~F"""
+    <div class={merge([
+      "w-full max-w-sm inline-block transform rounded-xl bg-gohan align-middle shadow-moon-lg transition-all",
+      @class
+    ])}>
+      {#if slot_assigned?(:header)}
```
Comment (phgrey) [inline lib/moon/design/modal/panel.ex:19]: have you tried if it works the same without this check?

**Review by phgrey — APPROVED** [review]
I do really like this component! not sure we have to make it statefful. see one minor question for the slot.
good job!

File: lib/moon/design/modal/panel.ex (hunk 5 above)
Comment (tkullisaar) [reply lib/moon/design/modal/panel.ex]: Works without as well, thanks, removed check.

**Pull request merged**